config/initializers/secure_headers.rb

```diff
@@ -33,3 +33,12 @@
   #   ]
   # }
 end
+
+SecureHeaders::Configuration.override(:saml) do |config|
+  providers = YAML.load_file("#{Rails.root}/config/service_providers.yml")
```

Comment:
The headers are generated once and reused over every request. See https://github.com/twitter/secureheaders#named-overrides